doc/source/tutorial/signal.rst

```diff
@@ -610,6 +610,35 @@ stop-band attenuation of :math:`\approx 60` dB.
    >>> plt.grid()
    >>> plt.show()
 
+.. note::
+
+    It is important to note that the cutoffs for :func:`firwin` and :func:`iirfilter` 
+    are defined differently. For :func:`firwin`, the cutoff-frequency is at 
+    half-amplitude (i.e. -6dB). For :func:`iirfilter`, the cutoff is at half-power
+    (i.e. -3dB).
+
+    .. plot::
+        :alt: "This code generates an example plot displaying the differences in cutoff frequency between FIR and IIR filters. FIR filters have a cutoff frequency at half-amplitude, while IIR filter cutoffs are at half-power."
+
+        >>> import numpy as np
+        >>> from matplotlib import pyplot as plt
+        >>> from scipy import signal as sig
+
+        >>> fs = 16000
+        >>> b = sig.firwin(101, 2500, fs=fs)
+        >>> f, h_fft = sig.freqz(b, fs=fs)
+        >>> h_amp = 20 * np.log10(np.abs(h_fft))
+        >>> plt.figure()
+        >>> plt.plot(f, h_amp)
+        >>> plt.grid()
+
+        >>> b, a = sig.iirfilter(15, 2500, btype="low", fs=fs)
+        >>> f, h_fft = sig.freqz(b, a, fs=fs)
+        >>> h_amp = 20 * np.log10(np.abs(h_fft))
+        >>> plt.plot(f, h_amp)
+        >>> plt.axis([2100, 2900, -10, 2])
+        >>> plt.legend(["FIR", "IIR"])
```

Comment:
Let's modernize this a bit -- class-based is preferred over global state MPL nowadays, constrained layouts generally look better, and we might as well label the axes:
```suggestion
        >>> _, ax = plt.subplots(layout="constrained")
        >>> ax.plot(f, h_amp, label="FIR")
        >>> ax.grid(True)

        >>> b, a = sig.iirfilter(15, 2500, btype="low", fs=fs)
        >>> f, h_fft = sig.freqz(b, a, fs=fs)
        >>> h_amp = 20 * np.log10(np.abs(h_fft))
        >>> ax.plot(f, h_amp, label="IIR")
        >>> ax.set(xlim=[2100, 2900], ylim=[-10, 2])
        >>> ax.set(xlabel="Frequency (Hz)", ylabel="Amplitude (dB)")
```

Reply:
... I know some of the other code uses the old interface, but we might as well make incremental improvement with the new code here!

Reply:
That makes sense, is there anything else in here that needs to be changed?